File: models/Servicio.cs
namespace TallerMecanico.models;

public class Servicio
{
    public int Id { get; set; }
    public string Nombre { get; set; }
    public string Descripcion { get; set; }
    public decimal Precio { get; set; }
    public string Imagen { get;set; }
    // Relación muchos a muchos con Clientes
    public ICollection<ClienteServicio> ClienteServicios { get; set; } = new List<ClienteServicio>();

    public ICollection<Vehiculo> Vehiculos { get; set; }
    // Auditoría
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
    public DateTime? FechaBorrado { get; set; }
    public bool EstaBorrado { get; set; } = false;
}

File: AFront/src/app/Entity/Servicio.ts
import { Cliente } from './Cliente';
import { Vehiculo } from './Vehiculo';

export interface Servicio {
  id: number;
  nombre: string;
  descripcion: string;
  precio: number|null;
  clientes: Cliente[];
  vehiculos: Vehiculo[];
  fechaCreacion: Date;
  fechaActualizacion: Date | null;
  fechaBorrado: Date | null;
  estaBorrado: boolean;
  imagen:string;
}
